python/tank/authentication/shotgun_shared/saml2_sso.py

```diff
@@ -582,17 +587,150 @@ def on_sso_disable_renewal(self, event):
 ################################################################################
 
 
-def _decode_cookies(app_cookies):
-    sg_cookies = []
-    if app_cookies:
-        sg_cookies = json.loads(base64.b64decode(app_cookies))
-        sg_cookies = [cookie.encode("utf-8") for cookie in sg_cookies]
-    return sg_cookies
+def _decode_cookies(encoded_cookies):
+    """
+    Extract the cookies from a base64 encoded string.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+
+    :returns: A SimpleCookie containing all the cookies.
+    """
+    cookies = SimpleCookie()
+    if encoded_cookies:
+        try:
+            decoded_cookies = base64.b64decode(encoded_cookies)
+            cookies.load(decoded_cookies)
+        except TypeError, e:
+            log.error("Unable to decode the cookies: %s" % e.message)
+    return cookies
+
+
+def _encode_cookies(cookies):
+    """
+    Extract the cookies from a base64 encoded string.
+
+    :param cookies: A Cookie.SimpleCookie instance representing the cookie jar.
+
+    :returns: An encoded string representing the cookie jar.
+    """
+    encoded_cookies = base64.b64encode(cookies.output())
+    return encoded_cookies
+
+
+def _get_shotgun_user_id(cookies):
+    """
+    Returns the id of the user in the shotgun instance, based on the cookies.
+
+    :param cookies: A Cookie.SimpleCookie instance representing the cookie jar.
+
+    :returns: A string user id value, or None.
+    """
+    user_id = None
+    for cookie in cookies:
+        if cookie.startswith("shotgun_sso_session_userid_u"):
+            if user_id is not None:
+                # Should we find multiple cookies with the same prefix, it means
+                # that we are using cookies from a multi-session environment. We
+                # have no way to identify the proper user id in the lot.
+                raise Saml2SssoMultiSessionNotSupportedError('Multi-session not supported')
+            user_id = cookie[28:]
+    return user_id
+
+
+def _get_cookie_from_prefix(encoded_cookies, cookie_prefix):
+    """
+    Returns a cookie value based on a prefix to which we will append the user id.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+    :param cookie_prefix: The prefix of the cookie name.
+
+    :returns: A string of the cookie value, or None.
+    """
+    value = None
+    cookies = _decode_cookies(encoded_cookies)
+    key = "%s%s" % (cookie_prefix, _get_shotgun_user_id(cookies))
+    if key in cookies:
+        value = cookies[key].value
+    return value
+
+
+def is_sso_enabled(encoded_cookies):
+    """
+    Indicate if SSO is being used from the Shotgun cookies.
+
+    :param encoded_cookies: An encoded string representing the cookie jar.
+
+    :returns: True or False
+    """
+    cookies = _decode_cookies(encoded_cookies)
+    return _get_shotgun_user_id(cookies) is not None
+
+
+def get_saml_claims_expiration(encoded_cookies):
```

Comment:
should we have a cookie class instead of all these utility methods all taking a `encoded_cookies` parameter? Seems like that could be cleaner.